src/Microsoft.AspNetCore.Razor.Language/CodeGeneration/CodeRenderingContext.cs

```diff
@@ -18,11 +19,15 @@ public abstract class CodeRenderingContext
         public abstract string DocumentKind { get; }
 
         public abstract ItemCollection Items { get; }
+        
+        public abstract IEnumerable<IntermediateNode> Ancestors { get; }
 
         public abstract IntermediateNodeWriter NodeWriter { get; }
 
         public abstract RazorCodeGenerationOptions Options { get; }
 
+        public abstract IntermediateNode Parent { get; }
```

Comment:
Not a ton of value in this being abstract. Could just introspect on our ancestors.